src/rgw/rgw_rest_log.cc

```diff
@@ -489,11 +521,12 @@ void RGWOp_BILog_Info::execute(optional_yield y) {
     }
   }
   map<RGWObjCategory, RGWStorageStats> stats;
-  int ret =  store->getRados()->get_bucket_stats(bucket_info, shard_id, &bucket_ver, &master_ver, stats, &max_marker, &syncstopped);
+  int ret =  store->getRados()->get_bucket_stats(bucket_info, -1, latest_gen, &bucket_ver, &master_ver, stats, &max_marker, &syncstopped);
```

Comment:
`get_bucket_stats()` is being called with a log generation here (`latest_gen = bucket_info.layout.logs.back().gen`), but `open_bucket_index()` seems to be using that as an index generation:
```c++
get_bucket_index_objects(bucket_oid_base, bucket_info.layout.current_index.layout.normal.num_shards, gen_id, bucket_objs, shard_id);
```

eventually i think we want to split `get_bucket_stats()` into two separate APIs - one to get bucket usage statistics that targets an index layout, and another to get the bilog markers associated with a log layout. but for now (while we can assume that log layouts are InIndex), `get_bucket_stats()` (along with `cls_bucket_head()` and `open_bucket_index()`) should require a `bucket_index_layout_generation`

so with `get_bucket_stats()` tied to an index layout, this `BILog_Info` API would transform the latest log layout into its corresponding index layout like:
```c++
const auto& latest_log = bucket_info.layout.logs.back();
ceph_assert(latest_log.layout.type == BucketLogType::InIndex); // only type supported

// convert InIndex log layout to its index layout
bucket_index_layout_generation index;
index.gen = latest_log.layout.in_index.gen;
index.layout.normal = latest_log.layout.in_index.layout;
```
and pass that `index` to `get_bucket_stats()`

Reply:
ok. I have used latest_log for all other occurences of get_bucket_stats() as well.